### PR TITLE
[MSPAINT] Add nearlyEqualPoints and use it

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Refactoring...
JIRA issue: [CORE-17931](https://jira.reactos.org/browse/CORE-17931)

## Proposed changes

- Add `nearlyEqualPoints` helper function and use it.
- The `nearlyEqualPoints`  function uses `GetSystemMetrics(SM_CXDRAG)` and `GetSystemMetrics(SM_CYDRAG)`.

### CODE REVIEW

File: base/applications/mspaint/mouse.cpp [hunk 1]
```diff
@@ -50,6 +51,13 @@ roundTo8Directions(LONG x0, LONG y0, LONG& x1, LONG& y1)
     }
 }
 
+BOOL nearlyEqualPoints(INT x0, INT y0, INT x1, INT y1)
+{
+    INT cxThreshold = toolsModel.GetLineWidth() + GetSystemMetrics(SM_CXDRAG);
```
Comment (roybaer) [inline base/applications/mspaint/mouse.cpp:56]: Wouldn't the GetSystemMetrics result have to be scaled to the current zoom level?

**Review by roybaer — COMMENTED** [review]
I don't know what the default values for SM_CXDRAG and SM_CYDRAG are, but they are certainly adjustable.
I'd say that this change makes sense with the small exception that the drag distance -- and only that -- should not be affected by zooming.

File: base/applications/mspaint/mouse.cpp (hunk 1 above)
Comment (katahiromz) [reply base/applications/mspaint/mouse.cpp:56]: The line width can be zoomed. The drag distance can't be.

Comment (katahiromz) [reply base/applications/mspaint/mouse.cpp]: OK. Fixed.

Comment (roybaer) [reply base/applications/mspaint/mouse.cpp]: Wait. Don't you call nearlyEqualPoints with transformed (zoomed) coordinates? Wouldn't it have to be the other way around, then?
I.e. something like UnZoomed(GetSystemMetrics(....))?

Comment (katahiromz) [reply base/applications/mspaint/mouse.cpp:56]: Umm, Reverted. `x0`, `y0`, `x1`, `y1` are unzoomed values. So, we have to calculate the threshold values in unzoomed values.

Comment (katahiromz) [reply base/applications/mspaint/mouse.cpp:56]: `GetSystemMetrics(SM_CXDRAG)` is already an unzoomed value.

Comment (roybaer) [reply base/applications/mspaint/mouse.cpp:56]: Maybe the term "unzoomed" is a bit to ambiguous. I was under the assumption that nearlyEqualPoints gets called with coordinates in the image's coordinate system, whereas GetSystemMetrics uses the screen's coordinate system, which means that one of them has to be scaled. Is this assumption correct? If both are in the screen's coordinate system, the line width would have to be scaled, though.

Comment (katahiromz) [reply base/applications/mspaint/mouse.cpp:56]: In the screen coordinate system, the pixel and the coordinate are 1:1. `GetSystemMetrics(SM_CXDRAG)` is also 1:1. This `nearlyEqualPoints` takes the image's coordinates.

Comment (katahiromz) [reply base/applications/mspaint/mouse.cpp:56]: It cannot get the accurate scaled coordinates from the image's coordinates.

Comment (katahiromz) [reply base/applications/mspaint/mouse.cpp]: Applied `UnZoomed` to `cxDrag` and `cyDrag`.
image
LGTM.

**Review by roybaer — APPROVED** [review]

**Pull request merged**